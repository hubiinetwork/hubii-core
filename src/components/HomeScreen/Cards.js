import styled from 'styled-components';

export default styled.div`
  display: flex;
  margin-top: 4rem;
  min-width: 70vw;
  max-width: 70rem;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center
`;
