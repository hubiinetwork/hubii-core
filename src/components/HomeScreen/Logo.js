import styled from 'styled-components';

export default styled.img`
  width: 30rem;
`;
