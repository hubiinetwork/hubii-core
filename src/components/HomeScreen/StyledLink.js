import styled from 'styled-components';
import { Link } from 'react-router-dom';

export default styled(Link)`
  display: flex;
  flex: 1;
  margin: 1rem;
`;
