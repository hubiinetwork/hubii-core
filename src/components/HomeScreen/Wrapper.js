import styled from 'styled-components';

export default styled.div`
  display: flex;
  flex: 1;
  padding: 0 3rem;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: ${(props) => props.theme.palette.primary1};
  overflow-x: hidden;
  overflow-y: scroll;
`;
