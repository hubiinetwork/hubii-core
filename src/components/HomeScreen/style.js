import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const Cards = styled.div`
  display: flex;
  margin-top: 4rem;
  max-width: 70rem;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center
`;

export const Logo = styled.img`
  width: 30rem;
`;

export const StyledLink = styled(Link)`
  display: flex;
  flex: 1;
  margin: 1rem;
`;

export const Wrapper = styled.div`
  display: flex;
  flex: 1;
  padding: 0 3rem;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: ${(props) => props.theme.palette.primary1};
  overflow-x: hidden;
  overflow-y: scroll;
`;
