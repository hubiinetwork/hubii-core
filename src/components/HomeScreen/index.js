/**
 *
 * HomeScreen
 *
 */

import React from 'react';
import DashboardCard from 'components/DashboardCard';

import { getAbsolutePath } from 'utils/electron';
import StyledLink from './StyledLink';
import Wrapper from './Wrapper';
import Logo from './Logo';
import Cards from './Cards';
function HomeScreen() {
  return (
    <Wrapper>
      <Logo src={getAbsolutePath('public/images/hubii-core-logo-wtext.svg')} />
      <Cards>
        <StyledLink to="/wallets">
          <DashboardCard iconSrc={getAbsolutePath('public/images/wallet-icon-green.png')} title="My wallets" />
        </StyledLink>
        <StyledLink to="/dex">
          <DashboardCard iconSrc={getAbsolutePath('public/images/dex-icon-green.png')} title="DEX" />
        </StyledLink>
      </Cards>
    </Wrapper>
  );
}

export default HomeScreen;
