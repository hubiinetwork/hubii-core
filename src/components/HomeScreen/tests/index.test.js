import React from 'react';
import { shallow } from 'enzyme';

import HomeScreen from '../index';

describe('<HomeScreen />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <HomeScreen />);
    expect(wrapper).toMatchSnapshot();
  });
});
