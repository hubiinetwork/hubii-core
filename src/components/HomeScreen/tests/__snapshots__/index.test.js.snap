// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeScreen
    intl={
        Object {
            "formatMessage": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Logo
          src="/public/images/hubii-core-logo-wtext.svg"
/>,
        <Cards>
          <StyledLink
                    to="/wallets"
          >
                    <DashboardCard
                              iconSrc="/public/images/wallet-icon-green.png"
                              title="my_wallets"
                    />
          </StyledLink>
          <StyledLink
                    to="/nahmii/airdriip-registration"
          >
                    <DashboardCard
                              iconSrc="/public/images/nahmii-token-green.png"
                              title="my_nahmii"
                    />
          </StyledLink>
          <StyledLink
                    to="/dex"
          >
                    <DashboardCard
                              iconSrc="/public/images/dex-icon-green.png"
                              title="DEX"
                    />
          </StyledLink>
          <StyledLink
                    to="/settings"
          >
                    <DashboardCard
                              iconType="setting"
                              title="settings"
                    />
          </StyledLink>
</Cards>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "src": "/public/images/hubii-core-logo-wtext.svg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <StyledLink
              to="/wallets"
>
              <DashboardCard
                            iconSrc="/public/images/wallet-icon-green.png"
                            title="my_wallets"
              />
</StyledLink>,
            <StyledLink
              to="/nahmii/airdriip-registration"
>
              <DashboardCard
                            iconSrc="/public/images/nahmii-token-green.png"
                            title="my_nahmii"
              />
</StyledLink>,
            <StyledLink
              to="/dex"
>
              <DashboardCard
                            iconSrc="/public/images/dex-icon-green.png"
                            title="DEX"
              />
</StyledLink>,
            <StyledLink
              to="/settings"
>
              <DashboardCard
                            iconType="setting"
                            title="settings"
              />
</StyledLink>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <DashboardCard
                iconSrc="/public/images/wallet-icon-green.png"
                title="my_wallets"
/>,
              "to": "/wallets",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "iconSrc": "/public/images/wallet-icon-green.png",
                "title": "my_wallets",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <DashboardCard
                iconSrc="/public/images/nahmii-token-green.png"
                title="my_nahmii"
/>,
              "to": "/nahmii/airdriip-registration",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "iconSrc": "/public/images/nahmii-token-green.png",
                "title": "my_nahmii",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <DashboardCard
                iconSrc="/public/images/dex-icon-green.png"
                title="DEX"
/>,
              "to": "/dex",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "iconSrc": "/public/images/dex-icon-green.png",
                "title": "DEX",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <DashboardCard
                iconType="setting"
                title="settings"
/>,
              "to": "/settings",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "iconType": "setting",
                "title": "settings",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Logo
            src="/public/images/hubii-core-logo-wtext.svg"
/>,
          <Cards>
            <StyledLink
                        to="/wallets"
            >
                        <DashboardCard
                                    iconSrc="/public/images/wallet-icon-green.png"
                                    title="my_wallets"
                        />
            </StyledLink>
            <StyledLink
                        to="/nahmii/airdriip-registration"
            >
                        <DashboardCard
                                    iconSrc="/public/images/nahmii-token-green.png"
                                    title="my_nahmii"
                        />
            </StyledLink>
            <StyledLink
                        to="/dex"
            >
                        <DashboardCard
                                    iconSrc="/public/images/dex-icon-green.png"
                                    title="DEX"
                        />
            </StyledLink>
            <StyledLink
                        to="/settings"
            >
                        <DashboardCard
                                    iconType="setting"
                                    title="settings"
                        />
            </StyledLink>
</Cards>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "src": "/public/images/hubii-core-logo-wtext.svg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <StyledLink
                to="/wallets"
>
                <DashboardCard
                                iconSrc="/public/images/wallet-icon-green.png"
                                title="my_wallets"
                />
</StyledLink>,
              <StyledLink
                to="/nahmii/airdriip-registration"
>
                <DashboardCard
                                iconSrc="/public/images/nahmii-token-green.png"
                                title="my_nahmii"
                />
</StyledLink>,
              <StyledLink
                to="/dex"
>
                <DashboardCard
                                iconSrc="/public/images/dex-icon-green.png"
                                title="DEX"
                />
</StyledLink>,
              <StyledLink
                to="/settings"
>
                <DashboardCard
                                iconType="setting"
                                title="settings"
                />
</StyledLink>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <DashboardCard
                  iconSrc="/public/images/wallet-icon-green.png"
                  title="my_wallets"
/>,
                "to": "/wallets",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "iconSrc": "/public/images/wallet-icon-green.png",
                  "title": "my_wallets",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <DashboardCard
                  iconSrc="/public/images/nahmii-token-green.png"
                  title="my_nahmii"
/>,
                "to": "/nahmii/airdriip-registration",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "iconSrc": "/public/images/nahmii-token-green.png",
                  "title": "my_nahmii",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <DashboardCard
                  iconSrc="/public/images/dex-icon-green.png"
                  title="DEX"
/>,
                "to": "/dex",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "iconSrc": "/public/images/dex-icon-green.png",
                  "title": "DEX",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <DashboardCard
                  iconType="setting"
                  title="settings"
/>,
                "to": "/settings",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "iconType": "setting",
                  "title": "settings",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
